src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -870,7 +873,19 @@ private void mapObservationUnitRows(
 		return allVariables;
 	}
 
-	private void addLocationIdVariable(final List<MeasurementVariable> environmentDetailAndConditionVariables) {
+	void addLocationVariables(final List<MeasurementVariable> environmentDetailAndConditionVariables) {
+		// check if LOCATION_ABBR already exists in the study, add if not present
+		final OptionalInt indexOfLocationAbbr = IntStream.range(0, environmentDetailAndConditionVariables.size())
+			.filter(i -> environmentDetailAndConditionVariables.get(i).getAlias().equals(TermId.LOCATION_ABBR.name()))
```

Comment:
@cheryleigh Users can define an alias for the environment detail variables so this checking might fail, use the cvtermid for checking if the variable is present.